File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [utterance]
  (cond
    (str/blank? utterance)                 "Fine. Be that way!"
    (and (re-matches #"[^a-z]+" utterance)
         (re-find #"[A-Z]" utterance))     "Whoa, chill out!"
    (.endsWith utterance "?")              "Sure."
    :else                                  "Whatever."))

File: scala/bob/src/main/scala/bob.scala
class Bob {
  def hey(utterance: String) = {
    if (utterance matches "\\s*") {
      "Fine. Be that way!"
    } else if ((utterance matches ".*[A-Z].*") &&
               !(utterance matches ".*[a-z].*")) {
      "Whoa, chill out!"
    } else if (utterance endsWith "?") {
      "Sure."
    } else {
      "Whatever."
    }
  }
}
